homeassistant/components/acmeda/base.py

```diff
@@ -56,10 +56,10 @@ async def async_added_to_hass(self) -> None:
 
     async def async_will_remove_from_hass(self) -> None:
         """Entity being removed from hass."""
-        self.roller.callback_unsubscribe(self.notify_update)
+        self.roller.callback_unsubscribe(self.async_notify_update)
 
     @callback
```

Comment:
If I remember correctly, callbacks can't be async

Reply:
Within the library, the callback resolver looks at the callback function and if it is a coroutine (async) creates a task on the event loop, otherwise it runs the callback on the executor.

Reply:
I'll wait for bdraco on this one, since the next patch will be out next friday, we don't have a rush to merge :)

Reply:
This will work but its very inefficient since creating a task is an order of magnitude more latent than executing a callback.

In core we have a decorator, `@callback` to tell core how to run a callback.

If its decorated with `@callback` it runs in the loop, if not it runs in the executor

https://github.com/home-assistant/core/blob/54052792738e62366b2e07ad6dfe3b1d9049b0c1/homeassistant/core.py#L247
https://github.com/home-assistant/core/blob/54052792738e62366b2e07ad6dfe3b1d9049b0c1/homeassistant/core.py#L253
https://github.com/home-assistant/core/blob/54052792738e62366b2e07ad6dfe3b1d9049b0c1/homeassistant/core.py#L387

In the library you could do something similar, however if you are always running everything in the same loop there is likely no need to ever run anything in the executor and you could run the callbacks directly in the library and remove the executor code.